tests/bot/cogs/test_antimalware.py

```diff
@@ -0,0 +1,159 @@
+import unittest
+from unittest.mock import AsyncMock, Mock
+
+from discord import NotFound
+
+from bot.cogs import antimalware
+from bot.constants import AntiMalware as AntiMalwareConfig, Channels, STAFF_ROLES
+from tests.helpers import MockAttachment, MockBot, MockMessage, MockRole
+
+MODULE = "bot.cogs.antimalware"
+
+
+class AntiMalwareCogTests(unittest.IsolatedAsyncioTestCase):
+    """Test the AntiMalware cog."""
+
+    def setUp(self):
+        """Sets up fresh objects for each test."""
+        self.bot = MockBot()
+        self.cog = antimalware.AntiMalware(self.bot)
+        self.message = MockMessage()
+        AntiMalwareConfig.whitelist = [".first", ".second", ".third"]
```

Comment:
Why are you monkeypatching this directly instead of using `unittest.mock.patch`? The downside of monkeypatching directly is that the monkeypatch will stay behind, as the object in memory is mutated. While `unittest.mock.patch` does the same, in principle, it restores the original after the fact. By providing an argument for `new`, we can determine what it should be patched with and it makes sure that the mock doesn't get passed as an attribute to test methods.

You could use `unittest.mock.patch` as a class decorator to patch this for the entire test case class and it will be cleaned up after the class ends. In this case, you're mutating that object in memory, which may result in unexpected values if other files/tests access it after this line of code has run. Another option would be to just patch it for the methods that rely on it. If there's more than one and the patch value is stable (which its locating in the `setUp` method implies), I'd just go for a class decorator.

```py
from unittest.mock import patch

MODULE = "bot.cogs.antimalware"

@patch(f"{MODULE}.AntiMalwareConfig.whitelist", new=[".first", ".second", ".third"])
class AntiMalwareCogTests(unittest.IsolatedAsyncioTestCase):
    """Test the AntiMalware cog."""
```

Reply:
I had not considered this, pushed the changes.